### PR TITLE
PTNFLY-452 List View:Expand Row - PatternFly Core

### PR DESCRIPTION
**Pull request by junezhang**
## Description

Implemented two kinds of expand page for list view.
https://patternfly.atlassian.net/browse/PTNFLY-452

Demo Page
- Simple Expansion - [http://devops-ucd.rhcloud.com/June/patternfly/PTNFLY-452/dist/tests/list-view-simple-expansion.html](http://devops-ucd.rhcloud.com/June/patternfly/PTNFLY-452/dist/tests/list-view-simple-expansion.html)
- Compound Expansion - [http://devops-ucd.rhcloud.com/June/patternfly/PTNFLY-452/dist/tests/list-view-compound-expansion.html](http://devops-ucd.rhcloud.com/June/patternfly/PTNFLY-452/dist/tests/list-view-compound-expansion.html)

New flies for simply expansion
tests/pages/_includes/widgets/list-view/list-view-page-simple-expansion.html
tests/pages/_includes/widgets/list-view/list-view-standard-rows-simple-expansion.html
tests/pages/list-view-simple-expansion.html

New files for compound expansion
tests/pages/_includes/widgets/list-view/list-view-page-compound-expansion.html
tests/pages/_includes/widgets/list-view/list-view-standard-rows-compound-expansion.html
tests/pages/list-view-compound-expansion.html

### CODE REVIEW

File: less/list-view.less [hunk 1]
```diff
@@ -237,3 +237,87 @@
 .list-view-pf-view {
   margin-top: 30px;
 }
+.list-group-item-simple-expand, .list-group-item-compound-expand {
+  &.active {
+    border: solid 1px @gray-light !important;
+    box-shadow: 0 0 0 #e5e5e5, 0 0 2px rgba(0, 0, 0, .12), 0 2px 4px rgba(0, 0, 0, .24);
+    color: @list-group-link-color;
```
Comment (jeff-phillips-18) [inline less/list-view.less]: This looks like a very marginal color change. Worth doing?

File: less/list-view.less [hunk 2]
```diff
@@ -237,3 +237,87 @@
...
+    margin-top: -1px;
+  }
+  &.active .list-group-item {
+    background-color: @list-view-hover-bg;
+    border-top: 1px solid @gray-light !important;
```
Comment (jeff-phillips-18) [inline less/list-view.less]: non-expanded active items have a blue border, why not keep that color when expanded? It is harder to tell if an item is active when expanded.

**Comment by andresgalante** [comment]
Hi @junezhang sorry it took me so long to review this PR.

I think we should approach the expansion box in a different way. I don't like that it's a separate div outside the row itself.

I'd make it a div inside `list-group-item`. You'll probably need to set the `order` of that div to 3 and make it `flex` expand the 100% of the row.

Also, if you are going to use bootstrap grid, please for the correct structure with rows and cols and never use a clear fix as an empty div.

Please let me know if you have any question.

Thanks!

**Comment by junezhang** [comment]
@andresgalante 

Thanks for your great suggestions. 
At beginning, I also wanted to put the panel into the .list-group-item, just tried some methods, they all didn't work.  So put it outside of .list-group-item. Currently, I have put the panel into .list-group-item with flex value. Also added some value to .list-group-item as following:
- flex-wrap: wrap, it will let the panel into the second row. 
- remove the list-group-item padding, because if we add padding to .list-group-item, then the panel can't be display in width 100%.
- As list-group-item padding removed, so need to add padding to checkbox and the right action buttons.

One question here: We put the panel into .list-group-item, then we will have two rows, but the first row not in an independent 'div', so it's difficult to bind event to the first row. For example, in simple expansion page, we need to click the first row to expand the panel, but like the below picture, we can't bind event to one specific ‘div’, because there are mutilple different blocks: A, B, C, D. 
So still need your help.

list-group-item

Thanks
June

**Comment by andresgalante** [comment]
Hi June,

I want to keep the structure of the list group. So each line should be a list group item. It's ok to have an extra div to generate a clicable area, but please keep it inside each item.

Also, please review bootstrap grid system, you don't need to use an empty clearfix, each row has a clearfix.

Thanks

**Comment by junezhang** [comment]
@andresgalante As this PR has conflicts, so I recreated a PR for it.
https://github.com/patternfly/patternfly/pull/487

**Pull request closed without merging**